lib/src/noop_span.dart

```diff
@@ -40,7 +37,7 @@ class NoopSpan implements Span {
   void finish({DateTime finishTime}) {}
 
   @override
-  void log(String event, {DateTime timestamp, Object payload}) {}
+  void log(LogData log) {}
```

Comment:
If we want to avoid breaking changes, could we instead deprecate the existing `log` and add a `logv2? I think the other breaking changes could be mitigated similarly by using versioned APIs.